### DISCUSSION TITLE
CLI argument to set http headers for the dev server

### DISCUSSION THREAD

**Discussion by bgirschig** [parcel-bundler/parcel#8608] [Ideas]
I'm facing an issue with my current project because I need to use a sharedArrayBuffer, which [requires](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer#security_requirements) some specific CORS headers.
Parcel's dev server does not add these headers by default and It doesn't look like there is an option to add them manually.

I can solve this in two ways:
1. Hard-coding the headers in the [setHeaders](https://github.com/parcel-bundler/parcel/blob/de1569572ab681bb6e90d2b330d002b5df7947ea/packages/reporters/dev-server/src/Server.js#L35) method (Easiest, but likely to cause side effects)
2. Adding a command-line argument: --header. For example in my case, I could start my dev server with
```
parcel index.html \
   --header "Cross-Origin-Opener-Policy: same-origin" \
   --header "Cross-Origin-Opener-Policy: same-origin"
```

So my questions are:
- If I make a CL with either of these, is it likely to be accepted ?
- Does the `--header` parameter name make sense ? Should it be `--headers` (to match webpack), or something else ?
- Reading [contributing.md](https://github.com/parcel-bundler/parcel/blob/v2/CONTRIBUTING.md#releasing-a-new-version), it looks like anyone can release a new version. Is that really the case (More broadly, how does the release process work) ?

**Comment by mischnic** [comment]
Does this work for you? https://github.com/parcel-bundler/parcel/discussions/6976#discussioncomment-1423396 (still isn't described in the docs unfortunately: https://github.com/parcel-bundler/website/issues/884)

We really want to avoid adding more and more CLI arguments...

**Comment by bgirschig** [comment]
I'll give it a try, thanks for the suggestion.

I understand not wanting to add arguments to the CLI.
Would the .parcelrc be a better place for this ? (I don't think there are any dev-server settings there, so I guess not...)

**Reply by mischnic** [reply]
No, compared to that CLI would be better, also because the CLI arguments correspond to this list of settings which already contains stuff like dev server port/host/https/... https://github.com/parcel-bundler/parcel/blob/c3bbe0a6160186f496ca2f9e9bead9376c0522f1/packages/core/types/index.js#L285-L330
